tests/models.py

```diff
@@ -0,0 +1,41 @@
+"""
+Test Application models used for `test_oauth2maker_cmd.py`
+"""
+from django.db import models
+
+
+class TestApplication(models.Model):
```

Comment:
@fredmajor why not just pick Application from existing packages?

Reply:
To make tests simple and clear. Different test-only Application classes act as test data for the command I am testing. And to decouple tests from production application classes, that may change in the future, what could break the tests for no real reason. For example, if I use a production Application class App1, and one day decide to add a `mandatory_field` to it, I would have to populate that `mandatory_field` to all command invocations in tests using that App1 class. Even though that would be totally unrelated to actual tests logic. I also test the current production Application classes, but in different files and those tests serve a different purpose.

Let me show you a difference in a test before introducing test-only classes and after, I believe this is easier to show than explain :)
```
# BEFORE
    @patch.object(TestedCommand, 'add_arguments', side_effect=test_add_arguments)
    @patch.object(TestedCommand, 'app_model')
    def test_create(self, model_mock, _add_arguments_mock):
        """
        Ensure Application gets created in the DB.
        """
        from .ide_test_compat import fake_app_name, fake_client_id, fake_client_secret, Application

        model_mock.return_value = Application
        app_name = fake_app_name()
        client_id = fake_client_id()
        client_secret = fake_client_secret()
        service_host = faker.url()
        token_uri = faker.url()

        cmd_args = [
            'oauth2_app_maker',
            '--name=%s' % app_name,
            '--client-id=%s' % client_id,
            '--authorization-grant-type=client-credentials',
            '--client-secret=%s' % client_secret,
            '--service-host=%s' % service_host,
            '--token-uri=%s' % token_uri
        ]
        call_command(*cmd_args)

        check_app = Application.objects.get(name=app_name)
        self.assertEqual(check_app.authorization_grant_type, 'client-credentials')
        self.assertEqual(check_app.client_id, client_id)
        self.assertEqual(check_app.client_secret, client_secret)
        self.assertEqual(check_app.service_host, service_host)
        self.assertEqual(check_app.token_uri, token_uri)
```
```
# AFTER
    @patch('oauth2_client.fetcher.timezone.now')
    @patch.object(TestedCommand, 'app_model')
    def test_create(self, model_mock, now_mock):
        """
        Ensure Application gets created in the DB.
        """
        from tests.ide_test_compat import fake_app_name, TestApplication

        model_mock.return_value = TestApplication
        now = datetime(2020, 1, 1, tzinfo=timezone('CET'))  # arbitrary datetime
        now_mock.return_value = now
        app_name = fake_app_name()

        cmd_args = [
            'app_maker_testing',
            '--name=%s' % app_name,
            '--required-property=hello',
            '--optional-property=world',
            ]
        call_command(*cmd_args)

        check_app = TestApplication.objects.get(name=app_name)
        self.assertEqual(check_app.required_property, "hello")
        self.assertEqual(check_app.optional_property, "world")
        self.assertEqual(check_app.created, now)
        self.assertIsNotNone(check_app.updated, now)
```